File: day1/day1.py
import numpy as np

with open("day1input.txt") as f:
    lines = f.readlines()

lines = [line.split() for line in lines]
list1, list2 = zip(*lines)

list1 = np.array(list(map(int, list1)))
list2 = np.array(list(map(int, list2)))

list1 = np.sort(list1, axis=0)
list2 = np.sort(list2, axis=0)

differences = np.array(list1) - np.array(list2)
print(sum(abs(differences)))




    

    

File: day1/day1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cmath> // abs
#include <algorithm> // sort

int main() {
    std::ifstream file("day1input.txt");
    int sum = 0;
    int num1, num2;
    std::vector<int> list1, list2;

    while (file >> num1 >> num2) {
        list1.push_back(num1);
        list2.push_back(num2);
    }
    sort(list1.begin(), list1.end());
    sort(list2.begin(), list2.end());

    for (size_t i = 0; i < list1.size(); i++) {
        sum += abs(list1[i] - list2[i]);
    }

    std::cout << sum << std::endl;
    return 0;
}